chain/chain/src/chain.rs

```diff
@@ -1876,18 +1876,39 @@ impl Chain {
                 }
             })
             .expect("results should resolve to a final outcome");
-        let receipts = outcomes.split_off(1);
-        let transaction = self
+        let receipts_outcome = outcomes.split_off(1);
+        let transaction: SignedTransactionView = self
             .store
             .get_transaction(hash)?
             .ok_or_else(|| ErrorKind::DBNotFoundErr(format!("Transaction {} is not found", hash)))?
             .clone()
             .into();
+        let transaction_outcome = outcomes.pop().unwrap();
+        let receipt_id_from_transaction = transaction_outcome.outcome.receipt_ids.get(0).cloned();
+        let is_local_receipt = transaction.signer_id == transaction.receiver_id;
```

Comment:
@evgenykuzyakov should we store local receipts in storage? That will be a bigger change

Reply:
Don't need to. So long as you follow the protocol upgrades here.